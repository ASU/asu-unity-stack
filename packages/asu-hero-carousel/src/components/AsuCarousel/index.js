import AsuCarousel from './AsuCarousel';

export default AsuCarousel;
