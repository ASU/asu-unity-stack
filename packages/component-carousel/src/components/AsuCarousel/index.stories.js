// @ts-check
/** @jsx h */
import { h, Fragment } from "preact";

import { AsuCarousel } from ".";

const myCarouselItems = [
  {
    id: 1,
    item: (
      <div className="card">
        <img
          className="card-img-top"
          src="https://source.unsplash.com/random/800x400?a=1"
          alt="Card image cap"
         />
        <div className="card-header">
          <h3 className="card-title">Card One</h3>
        </div>
        <div className="card-body">
          <p className="card-text">
            Body copy goes here. Limit to 5 lines max. Lorem ipsum dolor sit
            amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
            ut labore et dolore magna aliqua eiusmod tempo.{" "}
          </p>
        </div>
        <div className="card-button">
          <a href="#" className="btn btn-maroon">
            Button link here
          </a>
        </div>
      </div>
    ),
  },
  {
    id: 2,
    item: (
      <div className="card">
        <img
          className="card-img-top"
          src="https://source.unsplash.com/random/800x400?a=2"
          alt="Card image cap"
         />
        <div className="card-header">
          <h3 className="card-title">Card Two</h3>
        </div>
        <div className="card-body">
          <p className="card-text">
            Body copy goes here. Limit to 5 lines max. Lorem ipsum dolor sit
            amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
            ut labore et dolore magna aliqua eiusmod tempo.{" "}
          </p>
        </div>
        <div className="card-button">
          <a href="#" className="btn btn-maroon">
            Button link here
          </a>
        </div>
      </div>
    ),
  },
  {
    id: 3,
    item: (
      <div className="card">
        <img
          className="card-img-top"
          src="https://source.unsplash.com/random/800x400?a=3"
          alt="Card image cap"
         />
        <div className="card-header">
          <h3 className="card-title">Card Three</h3>
        </div>
        <div className="card-body">
          <p className="card-text">
            Body copy goes here. Limit to 5 lines max. Lorem ipsum dolor sit
            amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
            ut labore et dolore magna aliqua eiusmod tempo.{" "}
          </p>
        </div>
        <div className="card-button">
          <a href="#" className="btn btn-maroon">
            Button link here
          </a>
        </div>
      </div>
    ),
  },
  {
    id: 4,
    item: (
      <div className="card">
        <img
          className="card-img-top"
          src="https://source.unsplash.com/random/800x400?a=4"
          alt="Card image cap"
         />
        <div className="card-header">
          <h3 className="card-title">Card Four</h3>
        </div>
        <div className="card-body">
          <p className="card-text">
            Body copy goes here. Limit to 5 lines max. Lorem ipsum dolor sit
            amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
            ut labore et dolore magna aliqua eiusmod tempo.{" "}
          </p>
        </div>
        <div className="card-button">
          <a href="#" className="btn btn-maroon">
            Button link here
          </a>
        </div>
      </div>
    ),
  },
  {
    id: 5,
    item: (
      <div className="card">
        <img
          className="card-img-top"
          src="https://source.unsplash.com/random/800x400?a=5"
          alt="Card image cap"
         />
        <div className="card-header">
          <h3 className="card-title">Card Five</h3>
        </div>
        <div className="card-body">
          <p className="card-text">
            Body copy goes here. Limit to 5 lines max. Lorem ipsum dolor sit
            amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
            ut labore et dolore magna aliqua eiusmod tempo.{" "}
          </p>
        </div>
        <div className="card-button">
          <a href="#" className="btn btn-maroon">
            Button link here
          </a>
        </div>
      </div>
    ),
  },
];

export default {
  component: AsuCarousel,
  title: "Asu Carousel",
};

export const ThreeItemCarousel = () => (
  <AsuCarousel perView="3" carouselItems={myCarouselItems} />
);
export const TwoItemCarousel = () => (
  <AsuCarousel perView="2" carouselItems={myCarouselItems} />
);
export const OneItemCarousel = () => (
  <AsuCarousel perView="1" carouselItems={myCarouselItems} />
);
export const MultipleCarousels = () => (
  <>
    <AsuCarousel perView="3" carouselItems={myCarouselItems} />
    <AsuCarousel perView="1" carouselItems={myCarouselItems} />
    <AsuCarousel perView="2" carouselItems={myCarouselItems} />
  </>
);
export const BaseCarousel = () => (
  <AsuCarousel carouselItems={myCarouselItems} />
);
