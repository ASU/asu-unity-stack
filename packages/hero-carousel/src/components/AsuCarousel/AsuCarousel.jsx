import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  Carousel,
  CarouselItem,
  CarouselControl,
  CarouselIndicators,
  Util,
} from "reactstrap";
import { AsuHeroImage } from "../AsuHeroImage";
import AsuBgHeroImage from "../AsuBgHeroImage";
import bootstrap from "asu-web-standards-bootstrap4/dist/css/bootstrap-asu.min.css";

// Needs to be set for CSS modules.
// See https://github.com/reactstrap/reactstrap/issues/1049
Util.setGlobalCssModule(bootstrap);

const AsuCarousel = props => {
  const [activeIndex, setActiveIndex] = useState(0);
  const [animating, setAnimating] = useState(false);
  const items = props.items;

  const next = () => {
    if (animating) return;
    const nextIndex = activeIndex === items.length - 1 ? 0 : activeIndex + 1;
    setActiveIndex(nextIndex);
  };

  const previous = () => {
    if (animating) return;
    const nextIndex = activeIndex === 0 ? items.length - 1 : activeIndex - 1;
    setActiveIndex(nextIndex);
  };

  const goToIndex = newIndex => {
    if (animating) return;
    setActiveIndex(newIndex);
  };

  const slides = items.map((item, index) => {
    return (
      <CarouselItem
        tag="div"
        key={index}
        onExiting={() => setAnimating(true)}
        onExited={() => setAnimating(false)}
      >
        {props.variant == "background" ? (
          <AsuBgHeroImage {...item} />
        ) : (
          <AsuHeroImage {...item} />
        )}
      </CarouselItem>
    );
  });

  return (
    <Carousel
      activeIndex={activeIndex}
      next={next}
      previous={previous}
      cssModule={bootstrap}
      interval={props.interval}
      keyboard={props.keyboard}
      pause={props.pause}
      ride={props.ride}
      slide={props.slide}
      tag="div"
    >
      <CarouselIndicators
        items={props.items}
        activeIndex={activeIndex}
        onClickHandler={goToIndex}
      />
      {slides}
      <CarouselControl
        direction="prev"
        directionText="Previous"
        onClickHandler={previous}
      />
      <CarouselControl
        direction="next"
        directionText="Next"
        onClickHandler={next}
      />
    </Carousel>
  );
};

AsuCarousel.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  interval: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.bool,
  ]),
  keyboard: PropTypes.bool,
  /* If set to "hover", pauses the cycling of the carousel on mouseenter and resumes the cycling of the carousel on
   * mouseleave. If set to false, hovering over the carousel won't pause it. (default: "hover")
   */
  pause: PropTypes.oneOf(["hover", false]),
  // Autoplays the carousel after the user manually cycles the first item. If "carousel", autoplays the carousel on load.
  // This is how bootstrap defines it... I would prefer a bool named autoplay or something...
  ride: PropTypes.oneOf(["carousel"]),
  // controls whether the slide animation on the Carousel works or not
  slide: PropTypes.bool,
  variant: PropTypes.oneOf(["default", "background"]),
};

AsuCarousel.defaultProps = {
  interval: 5000,
  keyboard: true,
  pause: "hover",
  ride: "carousel",
  slide: true,
  variant: "default",
};

export default AsuCarousel;
