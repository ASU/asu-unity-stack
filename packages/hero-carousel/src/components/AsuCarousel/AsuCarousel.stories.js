import React from "react";
import AsuCarousel from "./index.js";
import { withKnobs, boolean, text, number } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";

const items = [
  {
    src:
      "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa1d%20text%20%7B%20fill%3A%23555%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa1d%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23777%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22285.921875%22%20y%3D%22218.3%22%3EFirst%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
    altText: "Slide 1",
    tagline: "Slide 1",
    title: "Slide 1 title",
    buttons: [{ text: "button 1", url: "https://asu.edu", color: "primary" }],
  },
  {
    src:
      "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa20%20text%20%7B%20fill%3A%23444%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa20%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23666%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22247.3203125%22%20y%3D%22218.3%22%3ESecond%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
    altText: "Slide 2",
    tagline: "A longer tagline with . some ? special * characters",
    title: "Slide 2 title",
    buttons: [
      { text: "button 2", url: "https://asu.edu", color: "secondary" },
      { text: "button 3", url: "https://asu.edu", color: "secondary" },
    ],
  },
  {
    src:
      "data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa21%20text%20%7B%20fill%3A%23333%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa21%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23555%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22277%22%20y%3D%22218.3%22%3EThird%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E",
    altText: "Slide 3",
    tagline: "Slide 3",
    title: "Slide 3 title",
  },
];

export default {
  component: AsuCarousel,
  title: "AsuCarousel",
  decorators: [withA11y, withKnobs],
};

export const basic = () => (
  <AsuCarousel
    items={items}
    interval={number("interval", 5000)}
    keyboard={boolean("keyboard", true)}
    pause={text("pause", "hover")}
    ride={text("ride", "carousel")}
    slide={boolean("slide", true)}
  />
);

export const backgroundVariant = () => (
  <AsuCarousel
    items={items}
    interval={number("interval", 5000)}
    keyboard={boolean("keyboard", true)}
    pause={text("pause", "hover")}
    ride={text("ride", "carousel")}
    slide={boolean("slide", true)}
    variant={text("variant", "background")}
  />
);
