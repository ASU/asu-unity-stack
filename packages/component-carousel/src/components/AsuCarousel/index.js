// @ts-check
import { BaseCarousel } from "../../core/components/BaseCarousel";

const AsuCarousel = BaseCarousel;

export { AsuCarousel };
