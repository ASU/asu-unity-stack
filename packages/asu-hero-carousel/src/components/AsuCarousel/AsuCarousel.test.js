import React from 'react'
import { render} from '@testing-library/react'
import AsuCarousel from './AsuCarousel'



describe('The AsuCarousel', () => {
    it('renders the component', () => {

      const {container} = render(<AsuCarousel  {...{items: [{title: "test"}]}} />);
            expect(container.firstChild).toMatchSnapshot()
    })
});
