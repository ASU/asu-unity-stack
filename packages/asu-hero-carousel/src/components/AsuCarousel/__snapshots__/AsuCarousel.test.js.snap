// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AsuCarousel renders the component 1`] = `
<div
  class="carousel slide"
>
  <ol
    class="carousel-indicators"
  >
    <li
      class="active"
    />
  </ol>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <div
        class="jumbotron jumbotron-fluid"
        style="padding: 0px;"
      >
        <div
          style="position: relative;"
        >
          <img
            class="w-100 d-block"
          />
          <div
            class="container-fluid"
            style="position: absolute; top: 0px;"
          >
            <h2
              class="heroTitle"
            >
              test
            </h2>
            <p
              class="heroP"
            />
            
          </div>
        </div>
      </div>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    role="button"
    style="cursor: pointer;"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="carousel-control-prev-icon"
    />
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    role="button"
    style="cursor: pointer;"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="carousel-control-next-icon"
    />
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
